package/MDAnalysis/coordinates/GRO.py

```diff
@@ -269,19 +275,35 @@ def write(self, selection, frame=None):
            resName and atomName are truncated to a maximum of 5 characters
         """
         # write() method that complies with the Trajectory API
-        u = selection.universe
-        if frame is not None:
-            u.trajectory[frame]  # advance to frame
-        else:
-            frame = u.trajectory.ts.frame
 
-        # make sure to use atoms (Issue 46)
-        atoms = selection.atoms
-        # can write from selection == Universe (Issue 49)
-        coordinates = atoms.positions
+        # if frame is not None:
+        #     u.trajectory[frame]  # advance to frame
+        # else:
+        #     frame = u.trajectory.ts.frame
+
+        try:
```

Comment:
`type(this) == that` is very literal and won't handle subclasses.  `AtomGroup` instances are subclasses of `groups.AtomGroup` as we customise the class for each Universe.  If you do `isinstance(atoms, groups.AtomGroup)` it should work.